src/ol/interaction/modify.js

```diff
@@ -490,6 +491,37 @@ ol.interaction.Modify.prototype.writeMultiPolygonGeometry_ = function(feature, g
 
 
 /**
+ * We convert a circle into two segments.  The segment at index {@link ol.MODIFY_SEGMENT_CIRCLE_CENTER_INDEX}
+ * is the circle's center (a point).  The segment at index {@link ol.MODIFY_SEGMENT_CIRCLE_CIRCUMFERENCE_INDEX}
+ * is the circumference, and is not a line segment.
+ *
+ * @param {ol.Feature} feature Feature.
+ * @param {ol.geom.Circle} geometry Geometry.
+ * @private
+ */
+ol.interaction.Modify.prototype.writeCircleGeometry_ = function(feature, geometry) {
+  var coordinates = geometry.getCenter();
+  var centerGeometry = new ol.geom.Point(coordinates, geometry.getLayout());
+  var centerSegmentData = /** @type {ol.ModifySegmentDataType} */ ({
+    feature: feature,
+    geometry: geometry,
+    index: ol.MODIFY_SEGMENT_CIRCLE_CENTER_INDEX,
+    segment: [coordinates, coordinates]
```

Comment:
Just because a segment has a starting and an end point. The circle is represented as two segments: a center and a circumference. This segment is the center, which starts and ends at the same point.